include/nbl/video/IAPIConnection.h

```diff
@@ -61,15 +60,38 @@ class NBL_API2 IAPIConnection : public core::IReferenceCounted
 
         const SFeatures& getEnabledFeatures() const { return m_enabledFeatures; }
 
-        const bool isRunningInRenderdoc() const { return m_rdoc_api; }
+        enum SDebuggerType
+        {
+            EDT_NONE,
+            EDT_RENDERDOC,
+            EDT_NGFX
+        };
+        const SDebuggerType isRunningInGraphicsDebugger() const { return m_debuggerType; }
         virtual bool startCapture() = 0;
         virtual bool endCapture() = 0;
 
     protected:
         IAPIConnection(const SFeatures& enabledFeatures);
 
         std::vector<std::unique_ptr<IPhysicalDevice>> m_physicalDevices;
+        SDebuggerType m_debuggerType;
         renderdoc_api_t* m_rdoc_api;
+
+        struct SNGFXIntegration 
+        {
+            SNGFXIntegration();
+
+            bool useNGFX = false;
+
+            bool injectNGFXToProcess();
+            bool executeNGFXCommand();
+            inline bool isAPILoaded() { return m_loaded; }
+        private:
+            const bool m_loaded;
+        };
+        using ngfx_api_t = SNGFXIntegration;
+        ngfx_api_t m_ngfx_api;
```

Comment:
typedef is not really necessary, the renderdoc thing is called `_t` cuase it has a C-API and that's what its called